be/src/udf/java/java_udf.cpp

```diff
@@ -76,6 +76,10 @@ static JNINativeMethod java_native_methods[] = {
 JVMFunctionHelper& JVMFunctionHelper::getInstance() {
     if (_env == nullptr) {
         _env = getJNIEnv();
+        if (_env == nullptr) {
+            const char* p = std::getenv("JAVA_HOME");
+            CHECK(p != nullptr) << "env 'JAVA_HOME' is not set";
```

Comment:
so here it will still crash instead of return errors?

Reply:
Please don't crash the whole backend process just because of a UDF, this error should be processed and return a error message to users.

Reply:
Yes.  We will crash here. It's same to the old behavior, but at least we know `JAVA_HOME` is not set.

Reply:
java udf has checked java home is set or not in the beginning. see this pr: https://github.com/StarRocks/starrocks/pull/10680